### PR TITLE
ci: bootstrap quickstart runners

### PR DESCRIPTION
**Pull request by coryan**
This PR introduces scripts to run the quickstart programs in all
generated libraries. It also changes the scaffold to generate these new
fragments in the generated `CMakeLists.txt` and `BUILD.bazel` files.

Part of the work for #7936

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8221)

### CODE REVIEW

**Comment by coryan** [comment]
I set the "do not merge" label because I want to talk about this over IM and/or the team meeting.

**Comment by coryan** [comment]
PTAL

File: cmake/quickstart-runner.cmake [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ~~~
+
+# A CMake script to run the quickstart programs.
+#
+# This script runs the "quickstart" program for a given library. We use CMake as
+# a scripting language because we know CMake to be installed and usable for both
```
Comment (devbww) [inline cmake/quickstart-runner.cmake:20]: "both" -v- three options (Windows, Linux, and macOS)?

File: cmake/quickstart-runner.cmake [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+# Windows, Linux, and macOS. At least while building with CMake.
```
Comment (devbww) [inline cmake/quickstart-runner.cmake:21]: "At least while building with CMake." should be part of the previous sentence ... perhaps parenthetical.

**Review by devbww — COMMENTED** [review]
Note: The PR description still mentions updating `BUILD.bazel` files.

File: cmake/quickstart-runner.cmake (hunk 2 above)
Comment (coryan) [reply cmake/quickstart-runner.cmake]: Thanks. Fixed both entries.

File: ci/cloudbuild/builds/shared.sh [hunk 3]
```diff
@@ -19,19 +19,28 @@ set -euo pipefail
...
 cmake --build cmake-out --target install
 
 # Tests the installed artifacts by building and running the quickstarts.
 quickstart::build_cmake_and_make "${INSTALL_PREFIX}"
 quickstart::run_cmake_and_make "${INSTALL_PREFIX}"
+
+# Verify the quickstart programs for generated libraries run. Note
+# that most of these run against production, and are therefore
+# integration tests with the usual flakiness issues.
+env -C cmake-out ctest --output-on-failure -L quickstart --parallel "$(nproc)"
```
Comment (devjgm) [inline ci/cloudbuild/builds/shared.sh:46]: I'm not sure what the output will look like in the logs, but you want want to add something like an `io::log_h2 "Running generated quickstarts"`, to separate the output from the output of the previously "run_cmake_and_make" target.

**Review by devjgm — APPROVED** [review]

File: ci/cloudbuild/builds/shared.sh (hunk 3 above)
Comment (coryan) [reply ci/cloudbuild/builds/shared.sh:47]: Done.

**Review by devjgm — APPROVED** [review]

**Pull request merged**